test/unit/host_discovered_test.rb

```diff
@@ -84,8 +84,33 @@ class HostDiscoveredTest < ActiveSupport::TestCase
     assert host.name.start_with?('mac')
   end
 
+  test 'discovered host can be searched in multiple taxonomies' do
+    org1 = FactoryGirl.create(:organization, :name => 'org1')
+    org2 = FactoryGirl.create(:organization, :name => 'org2')
+    org3 = FactoryGirl.create(:organization, :name => 'org3')
+    user = FactoryGirl.create(:user, :organizations => [org1, org2])
+    host1 = FactoryGirl.create(:host, :type => "Host::Discovered", :organization => org1)
+    host2 = FactoryGirl.create(:host, :type => "Host::Discovered", :organization => org2)
+    host3 = FactoryGirl.create(:host, :type => "Host::Discovered", :organization => org3)
```

Comment:
try `FactoryGirl.create(:discovered_host, :organization => org1)` it should work, definition is at https://github.com/theforeman/foreman_discovery/blob/develop/test/factories/discovery_rule_related.rb#L15